runtime/vam/expr/function/fields.go

```diff
@@ -0,0 +1,94 @@
+package function
+
+import (
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/vector"
+)
+
+// https://github.com/brimdata/zed/blob/main/docs/language/functions.md#fields
+type Fields struct {
+	zctx     *zed.Context
+	innerTyp *zed.TypeArray
+	outerTyp *zed.TypeArray
+}
+
+func NewFields(zctx *zed.Context) *Fields {
+	inner := zctx.LookupTypeArray(zed.TypeString)
+	return &Fields{
+		zctx:     zctx,
+		innerTyp: inner,
+		outerTyp: zctx.LookupTypeArray(inner),
+	}
+}
+
+func (f *Fields) Call(args ...vector.Any) vector.Any {
+	val := vector.Under(args[0])
+	switch typ := val.Type().(type) {
+	case *zed.TypeRecord:
+		paths := buildPath(typ, nil)
+		s := vector.NewStringEmpty(val.Len(), nil)
+		inOffs, outOffs := []uint32{0}, []uint32{0}
+		nulls := vector.NewBoolEmpty(val.Len(), nil)
+		for i := uint32(0); i < val.Len(); i++ {
+			if vector.IsNull(val, i) {
```

Comment:
runtime/sam/expr/function/fields.go doesn't treat a null record value any differently than a non-null record value. Is there a reason for doing that here?

Reply:
Good point, I'll change this.